axum-extra/src/routing/resource.rs

```diff
@@ -136,34 +125,6 @@ where
         self.route(&path, delete(handler))
     }
 
-    /// Nest another route at the "member level".
-    ///
-    /// The routes will be nested at `/{resource_name}/:{resource_name}_id`.
-    pub fn nest<T>(mut self, svc: T) -> Self
```

Comment:
This change seems unrelated to the matchit upgrade, how does it relate?

Reply:
`Resource::{nest, nest_collection}` would accept a `Service` and call `Router::nest`. If you passed a `Router` to `nest_collection` and also had a route for `edit` like so

```rust
Resource::named("users")
    .edit(|| async move {})
    .nest_collection(
        Router::new().route("/featured", get(|| async move {})),
    );
```

That would work because `Router::nest` would downcast the nested service to a router and move over the routes.

However now that we don't downcast anymore you're essentially doing this:

```rust
Router::new()
    .route("/users/:id/edit", get(|| async {}))
    .route("/users/*rest", Router::new());
```

Which is a conflict. `/users/1/edit` could match the edit route with `id = 1` or could match the wildcard with `rest = 1/edit`.

The same would happen previously if you didn't pass a `Router` but some opaque service.

It might be possible to work around this by complicating `Resource` a bunch but I kinda just wanna remove `Resource::nest` and `Resource::nest_collection`. Instead people can convert the `Resource` into a `Router` and add whatever additional routes they want:

```rust
let users = Resource::named("users").index(|| async { "users#index" });
let router = Router::from(users);
router.route("/users/:id/tweets", get(|| async {}));
```